[Bug]: no picture is shown on the map

### ⚠️ This issue respects the following points: ⚠️

- [x] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on Github _(I've searched it)_.
- [x] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [x] Maps app **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I completely reinstalled maps: I removed the app and the tables, and then the installation process recreated the tables. I ran maps:scan-photos, which inserted around 400000 jobs, which were then processed, but no picture is shown on the map. I know I have photos that could be displayed on the map because with 28.x Nextcloud version they were there, but not with the current installation I run, i.e. 30.0.6.2 (according to config.php).

### Steps to reproduce

1. completelly reinstall maps, including the tables recreation
2. check that everything is ok: occ maintenance:repair --include-expensive -vvv
3. run maps:scan-photos and check that the related jobs are created and processed
4. check the photos on the map
5. observe that no photo is displayed on the map

### Expected behavior

The photos are displayed on the map.

### Installation method

Manual installation

### Operating system

Debian/Ubuntu 24.10

### PHP engine version

8.3

### Web server

Nginx

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install of the server?

With Nextcloud 28.x it wasn't present but I don't remember when since what version the issue occured.

### Is this bug present after an update or on a fresh install of the app?

I manually reinstalled Maps on Nextcloud 30.0.6, but the issue was already present in the initial installation.

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [x] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "auth.bruteforce.protection.enabled": false,
        "maintenance_window_start": 100,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "adrhc.go.ro"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "default_phone_region": "RO",
        "logtimezone": "Europe\/Bucharest",
        "debug": false,
        "logfilemode": 432,
        "logfile": "\/home\/gigi\/apps\/log\/nextcloud.log",
        "loglevel": 1,
        "log_rotate_size": 10485760,
        "dbtype": "mysql",
        "version": "30.0.6.2",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "maintenance": false,
        "trashbin_retention_obligation": "auto",
        "overwriteprotocol": "https",
        "overwritehost": "adrhc.go.ro",
        "overwritewebroot": "\/nextcloud",
        "overwrite.cli.url": "https:\/\/adrhc.go.ro\/nextcloud",
        "preview_max_memory": 8192,
        "preview_max_filesize_image": 512,
        "preview_max_scale_factor": 5,
        "enabledPreviewProviders": [
            "OC\\Preview\\Illustrator",
            "OC\\Preview\\Movie",
            "OC\\Preview\\MSOffice2003",
            "OC\\Preview\\MSOffice2007",
            "OC\\Preview\\MSOfficeDoc",
            "OC\\Preview\\PDF",
            "OC\\Preview\\Photoshop",
            "OC\\Preview\\Postscript",
            "OC\\Preview\\StarOffice",
            "OC\\Preview\\SVG",
            "OC\\Preview\\TIFF",
            "OC\\Preview\\Font",
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\OpenDocument",
            "OC\\Preview\\Krita",
            "OC\\Preview\\EMF",
            "OC\\Preview\\HEIC"
        ],
        "metadata_max_filesize": 1024,
        "max_file_conversion_filesize": 1024,
        "defaultapp": "",
        "check_for_working_htaccess": false,
        "memories.exiftool": "\/home\/gigi\/apps\/opt\/php-pages\/nextcloud\/apps\/memories\/bin-ext\/exiftool-amd64-glibc",
        "memories.vod.path": "\/home\/gigi\/apps\/opt\/php-pages\/nextcloud\/apps\/memories\/bin-ext\/go-vod-amd64",
        "memories.vod.ffmpeg": "\/usr\/bin\/ffmpeg",
        "memories.vod.ffprobe": "\/usr\/bin\/ffprobe",
        "memories.db.triggers.fcu": true,
        "memories.vod.vaapi": true,
        "memories.gis_type": 1,
        "memories.index.path.blacklist": "^\\\/(@(Recycle|eaDir)|Documents|Notes|Talk|Templates|\\.trash)\\\/",
        "jpeg_quality": 100,
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "memories.vod.qf": 35,
        "DATA": "\/home\/gigi\/apps\/opt\/php-pages\/nextcloud-data"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 3.0.0
  - app_api: 4.0.6
  - bruteforcesettings: 3.0.0
  - camerarawpreviews: 0.8.6
  - circles: 30.0.0
  - cloud_federation_api: 1.13.0
  - comments: 1.20.1
  - contacts: 7.0.1
  - contactsinteraction: 1.11.0
  - dashboard: 7.10.0
  - dav: 1.31.1
  - drawio: 3.0.3
  - federatedfilesharing: 1.20.0
  - federation: 1.20.0
  - files: 2.2.0
  - files_downloadlimit: 3.0.0
  - files_external: 1.22.0
  - files_pdfviewer: 3.0.0
  - files_reminders: 1.3.0
  - files_sharing: 1.22.0
  - files_trashbin: 1.20.1
  - files_versions: 1.23.0
  - firstrunwizard: 3.0.0
  - logreader: 3.0.0
  - lookup_server_connector: 1.18.0
  - maps: 1.5.0
  - memories: 7.5.2
  - nextcloud_announcements: 2.0.0
  - notifications: 3.0.0
  - oauth2: 1.18.1
  - password_policy: 2.0.0
  - photos: 3.0.2
  - previewgenerator: 5.8.0
  - privacy: 2.0.0
  - provisioning_api: 1.20.0
  - recognize: 8.2.0
  - recommendations: 3.0.0
  - related_resources: 1.5.0
  - serverinfo: 2.0.0
  - settings: 1.13.0
  - sharebymail: 1.20.0
  - spreed: 20.1.4
  - support: 2.0.0
  - survey_client: 2.0.0
  - systemtags: 1.20.0
  - text: 4.1.0
  - theming: 2.6.0
  - twofactor_backupcodes: 1.19.0
  - updatenotification: 1.20.0
  - user_status: 1.10.0
  - viewer: 3.0.0
  - weather_status: 1.10.0
  - webhook_listeners: 1.1.0-dev
  - workflowengine: 2.12.0
Disabled:
  - admin_audit: 1.20.0
  - encryption: 2.18.0
  - notes: 4.11.0 (installed 4.11.0)
  - suspicious_login: 8.0.0 (installed 6.0.0)
  - twofactor_nextcloud_notification: 4.0.0
  - twofactor_totp: 12.0.0-dev
  - unsplash: 3.0.3 (installed 3.0.3)
  - user_ldap: 1.21.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
{"reqId":"r3pKYdPeJ9pFvKJvFCTP","level":2,"time":"2025-03-07T00:18:26+02:00","remoteAddr":"","user":"--","app":"cron","method":"","url":"/nextcloud/cron.php","message":"Cron job used more than 300 MB of ram after executing job OCA\\Recognize\\BackgroundJobs\\ClusterFacesJob (id: 1448269, arguments: {\"userId\":\"gigi\"}): 3.2 GB (before: 12.8 MB)","userAgent":"--","version":"30.0.6.2","data":{"app":"cron"}}
{"reqId":"27ZT6zHtJXmU8D1SPi0d","level":2,"time":"2025-03-07T00:29:57+02:00","remoteAddr":"","user":"--","app":"cron","method":"","url":"/nextcloud/cron.php","message":"Cron job used more than 300 MB of ram after executing job OCA\\Recognize\\BackgroundJobs\\ClusterFacesJob (id: 1448270, arguments: {\"userId\":\"gigi\"}): 3.2 GB (before: 12.8 MB)","userAgent":"--","version":"30.0.6.2","data":{"app":"cron"}}
{"reqId":"AokqqooWjtqlfQM6MMQX","level":1,"time":"2025-03-07T00:31:34+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"QLQLB6JHBW6RUvmdpvCk","level":1,"time":"2025-03-07T01:00:38+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"TfJTjdpf0QI7Nw05iZmN","level":1,"time":"2025-03-07T01:30:54+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"4CvADrciOTh3K2A4iJzs","level":1,"time":"2025-03-07T02:00:35+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"pUQkyiaZlAWZzHg68v8S","level":1,"time":"2025-03-07T02:15:40+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"6p73hgjumv5k45nwwELk","level":1,"time":"2025-03-07T02:45:32+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"0FgX6mjzkPDSt0RpMVuY","level":1,"time":"2025-03-07T03:00:33+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"BlmMfkfbQS4iCMYLg60G","level":1,"time":"2025-03-07T03:30:31+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"5HtLqVW6CgnwWKO57nEv","level":1,"time":"2025-03-07T04:00:33+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"QvlXQd8Z6NPyMoCmKBz7","level":1,"time":"2025-03-07T04:30:30+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"rJUTLhM2IRD7tw2dnE3t","level":1,"time":"2025-03-07T04:45:31+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"eh2d3Ek4rojWEue1FuTx","level":1,"time":"2025-03-07T05:15:31+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"Himmx7NSY7g2XRmuH59K","level":1,"time":"2025-03-07T05:45:30+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"tGQWu5fIwLvf9ZK6lVmL","level":1,"time":"2025-03-07T06:15:29+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"7tjgo4aYDwPPWqDxXGnk","level":1,"time":"2025-03-07T06:32:44+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"3kLM5KxfLYHgS5LBoeb3","level":1,"time":"2025-03-07T07:00:29+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"whD6svuzrotwLLDnVC3E","level":1,"time":"2025-03-07T07:30:27+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"QZMq7uPz6Rra0gvLzaiU","level":1,"time":"2025-03-07T07:45:31+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"3w2ur5bGlPkac8IGtsvz","level":1,"time":"2025-03-07T08:15:26+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"t4j3E4uy1QZbX2dyUGWe","level":1,"time":"2025-03-07T08:45:26+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"eLXix6bWD9YHMHF8SMLF","level":1,"time":"2025-03-07T09:00:28+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"FIm70WWruVE0m6vcj6Mg","level":1,"time":"2025-03-07T09:30:24+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"ON3zwninrVAWCmWhzi0W","level":1,"time":"2025-03-07T10:00:23+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"KTAU0WpOWUoAd0t0Qmt1","level":1,"time":"2025-03-07T10:15:30+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"1gWbYmkMpngf31zFTPeY","level":1,"time":"2025-03-07T10:40:22+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
{"reqId":"X3dGA8aci9dgahSr8JDK","level":1,"time":"2025-03-07T11:00:24+02:00","remoteAddr":"","user":"--","app":"memories","method":"","url":"/nextcloud/cron.php","message":"Indexing completed successfully","userAgent":"--","version":"30.0.6.2","data":{"app":"memories"}}
```

### Browser Console

```shell
App.vue:763 [Violation] Only request geolocation information in response to a user gesture.
sendPositionLoop @ App.vue:763
(anonymous) @ App.vue:771
2NotificationsApp-DV09BU4a.chunk.mjs:35 Got notification data, restoring default polling interval.
Navigated to https://adrhc.go.ro/nextcloud/index.php/apps/maps/
logger.chunk.mjs:2 [INFO] activity: Activity API registered {app: 'activity', uid: 'adr', level: 1}
Sidebar.js:19 OCA.Files.Sidebar initialized
comments-app.js:15 OCA.Comments.View initialized
ShareSearch.js:16 OCA.Sharing.ShareSearch initialized
ExternalLinkActions.js:16 OCA.Sharing.ExternalLinkActions initialized
ExternalShareActions.js:16 OCA.Sharing.ExternalShareActions initialized
previewUtils-BnCbNhwl.chunk.mjs:6 [INFO] viewer: No Files app found, viewer is now in standalone mode {app: 'viewer', uid: 'adr', level: 1, ocp: undefined, oca: undefined}
hook.js:608 [ERROR] viewer: Could not register handler {app: 'viewer', uid: 'adr', level: 1, error: "The handler doesn't have a valid component", handler: {…}}
overrideMethod @ hook.js:608
log @ previewUtils-BnCbNhwl.chunk.mjs:6
error @ previewUtils-BnCbNhwl.chunk.mjs:6
registerHandler @ viewer-init.mjs?v=15f14458-0:4788
(anonymous) @ register-viewer.js?v=15f14458-0:2Understand this errorAI
previewUtils-BnCbNhwl.chunk.mjs:6 [INFO] viewer: 5 viewer handlers registered {app: 'viewer', uid: 'adr', level: 1, handlers: Array(5)}
search.js:22 Initializing unified search plugin-filters from talk
session-heartbeat.js:83 session heartbeat polling started
index.mjs:48 [INFO] comments: Comments plugin registered for Activity sidebar action {app: 'comments', uid: 'adr', level: 1}
App.vue:763 [Violation] Only request geolocation information in response to a user gesture.
sendPositionLoop @ App.vue:763
created @ App.vue:630
hn @ vue.runtime.esm.js:3033
Vn @ vue.runtime.esm.js:4048
e._init @ vue.runtime.esm.js:5716
o @ vue.runtime.esm.js:5849
(anonymous) @ vue.runtime.esm.js:4564
init @ vue.runtime.esm.js:4426
(anonymous) @ vue.runtime.esm.js:6591
h @ vue.runtime.esm.js:6545
Dr.nodeOps @ vue.runtime.esm.js:7133
e._update @ vue.runtime.esm.js:3781
a @ vue.runtime.esm.js:3891
e.get @ vue.runtime.esm.js:3462
e @ vue.runtime.esm.js:3452
e @ vue.runtime.esm.js:3908
za.$mount @ vue.runtime.esm.js:8797
e._init @ vue.runtime.esm.js:5724
za @ vue.runtime.esm.js:5785
qo @ main.js:104
(anonymous) @ optionsController.js:45
leaflet-src.js:3760 [Violation] Only request geolocation information in response to a user gesture.
locate @ leaflet-src.js:3760
_activate @ L.Control.Locate.min.js:3
start @ L.Control.Locate.min.js:3
initLocControl @ Map.vue:639
onMapReady @ Map.vue:447
hn @ vue.runtime.esm.js:3033
n @ vue.runtime.esm.js:1832
hn @ vue.runtime.esm.js:3033
e.$emit @ vue.runtime.esm.js:3732
(anonymous) @ LMap.js:354
(anonymous) @ vue.runtime.esm.js:3159
yn @ vue.runtime.esm.js:3081
NotificationsApp-DV09BU4a.chunk.mjs:35 Registering notifications container as a menu
NotificationsApp-DV09BU4a.chunk.mjs:35 Notifications permissions granted
NotificationsApp-DV09BU4a.chunk.mjs:35 Polling interval updated to 30000
NotificationsApp-DV09BU4a.chunk.mjs:35 Started background fetcher as session_keepalive is enabled
NotificationsApp-DV09BU4a.chunk.mjs:35 Registering notifications container as a menu
NotificationsApp-DV09BU4a.chunk.mjs:35 Notifications permissions granted
NotificationsApp-DV09BU4a.chunk.mjs:35 Polling interval updated to 30000
NotificationsApp-DV09BU4a.chunk.mjs:35 Started background fetcher as session_keepalive is enabled
2NotificationsApp-DV09BU4a.chunk.mjs:35 Got notification data, restoring default polling interval.
App.vue:763 [Violation] Only request geolocation information in response to a user gesture.
sendPositionLoop @ App.vue:763
(anonymous) @ App.vue:771
```

### Additional info

I have plenty of RAM, so I configured php and Nextcloud to use as much as they need.
All my photos are visible in Photos and Memories applications.
`occ setupchecks -vvv` and `occ check -vvv` confirm everything is ok.

--- a/lib/Hooks/FileHooks.php
+++ b/lib/Hooks/FileHooks.php
@@ -16,11 +16,11 @@ use OC\Files\Filesystem;
 use OCA\Maps\Service\PhotofilesService;
 use OCA\Maps\Service\TracksService;
 use OCP\Files\FileInfo;
-use OCP\Files\IHomeStorage;
 use OCP\Files\IRootFolder;
 use OCP\Lock\ILockingProvider;
 use OCP\Share;
 use OCP\Util;
+use function OCP\Log\logger;
 
 /**
  * Handles files events
@@ -44,7 +44,8 @@ class FileHooks {
 
 	public function register() {
 		$fileWriteCallback = function (\OCP\Files\Node $node) {
-			if ($this->isUserNode($node) && $node->getSize() > 0) {
+			//logger('maps')->debug("Hook postWrite");
+			if ($node->getType() === FileInfo::TYPE_FILE && $this->isUserNode($node) && $node->getSize()) {
 				$path = $node->getPath();
 				if (!$this->lockingProvider->isLocked($path, ILockingProvider::LOCK_SHARED)
 					and !$this->lockingProvider->isLocked($path, ILockingProvider::LOCK_EXCLUSIVE)
@@ -59,6 +60,7 @@ class FileHooks {
 		$this->root->listen('\OC\Files', 'postWrite', $fileWriteCallback);
 
 		$fileDeletionCallback = function (\OCP\Files\Node $node) {
+			//logger('maps')->debug("Hook preDelete");
 			if ($this->isUserNode($node)) {
 				if ($node->getType() === FileInfo::TYPE_FOLDER) {
 					$this->photofilesService->deleteByFolder($node);
@@ -111,6 +113,7 @@ class FileHooks {
 	}
 
 	public function postShare($params) {
+		//logger('maps')->debug("Hook postShare");
 		if ($params['itemType'] === 'file') {
 			//$targetFilePath = $params['itemTarget'];
 			//$sourceUserId = $params['uidOwner'];
@@ -135,6 +138,7 @@ class FileHooks {
 	}
 
 	public function postUnShare($params) {
+		//logger('maps')->debug("Hook postUnShare");
 		if ($params['shareType'] === Share::SHARE_TYPE_USER) {
 			if ($params['itemType'] === 'file') {
 				$targetUserId = $params['shareWith'];
@@ -146,6 +150,7 @@ class FileHooks {
 	}
 
 	public function preUnShare($params) {
+		//logger('maps')->debug("Hook preUnShare");
 		if ($params['shareType'] === Share::SHARE_TYPE_USER) {
 			if ($params['itemType'] === 'folder') {
 				$targetUserId = $params['shareWith'];
@@ -176,7 +181,12 @@ class FileHooks {
 	}
 
 	private function isUserNode(\OCP\Files\Node $node): bool {
-		return $node->getStorage()->instanceOfStorage(IHomeStorage::class);
+		//return $node->getStorage()->instanceOfStorage("\OCP\Files\IHomeStorage")
+		$owner = $node->getStorage()->getOwner('');
+		if (! $owner) {
+			return false;
+		}
+		return str_starts_with($node->getPath(), '/' . $owner . '/');
 	}
 
 }

--- a/lib/Service/PhotofilesService.php
+++ b/lib/Service/PhotofilesService.php
@@ -96,14 +96,16 @@ class PhotofilesService {
 	// add the file for its owner and users that have access
 	// check if it's already in DB before adding
 	public function addByFile(Node $file) {
-		$ownerId = $file->getOwner()->getUID();
 		if ($this->isPhoto($file)) {
+			$ownerId = $file->getOwner()->getUID();
 			$this->addPhoto($file, $ownerId);
 			// is the file accessible to other users ?
 			$accesses = $this->shareManager->getAccessList($file);
-			foreach ($accesses['users'] as $uid) {
-				if ($uid !== $ownerId) {
-					$this->addPhoto($file, $uid);
+			if (array_key_exists('users', $accesses)) {
+				foreach ($accesses['users'] as $uid) {
+					if ($uid !== $ownerId) {
+						$this->addPhoto($file, $uid);
+					}
 				}
 			}
 			return true;

